dss/api/files.py

```diff
@@ -112,6 +116,25 @@ def get_helper(uuid: str, replica: Replica, version: str=None, token: str=None):
 def _verify_checkout(
         replica: Replica, token: typing.Optional[str], file_metadata: dict, blob_path: str,
 ) -> typing.Tuple[str, bool]:
+    try:
+        cloud_handle = Config.get_blobstore_handle(replica)
+        hca_handle = Config.get_hcablobstore_handle(replica)
+        checkout_file_key = get_dst_key(blob_path)
+        last_modified = cloud_handle.get_last_modified_date(replica.checkout_bucket, "files/{}".format(blob_path))
+        stale_before_date = (datetime.datetime.now(datetime.timezone.utc) -
+                             datetime.timedelta(days=int(os.environ['DSS_BLOB_PUBLIC_TTL_DAYS'])))
+
+        if last_modified > stale_before_date:
+            if hca_handle.verify_blob_checksum_from_dss_metadata(replica.checkout_bucket,
+                                                                 checkout_file_key,
+                                                                 file_metadata):
+                return "", True
+            else:
+                logger.error(
+                    f"Failed to verify file checksum during checkout: {replica.checkout_bucket}/{checkout_file_key}")
```

Comment:
Prefer: `Checksum verification failed.`

Failed to verify checksum could mean that you tried to verify the checksum and couldn't actually perform the operation _or_ the checksum verification failed.